src/ButtonAnchor.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+export default class ButtonAnchor extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+  render() {
+    let { href, role, tabIndex, disabled, style } = this.props;
+
+    if (disabled) {
+      tabIndex = -1;
+      style = { pointerEvents: 'none', ...style };
+    }
+
+    return (
+      <a
+        {...this.props}
+        style={style}
+        disabled={null}
+        // we want to make sure there is a href attribute on the node
+        // otherwise, the cursor incorrectly styled (except with role='button')
+        href={href || ''}
+        role={role || 'button'}
+        // http://getbootstrap.com/javascript/#dismiss-on-next-click
+        tabIndex={tabIndex || 0}
+        onClick={this.handleClick}
+      />
+    );
+  }
+
+  handleClick(event) {
+    let { disabled, onClick } = this.props;
+
+    event.preventDefault();
```

Comment:
yeah that was the intention, if then link is really a link then it shouldn't use this component.

In other words this is trying to encapsulate "Making a `<button>` but with an `<a>` tag for reasons of browsers being weird"

In the case of SafeAnchor use it will never prevent actual navigation because this only gets rendered when there is no `href`